x-pack/plugins/security/server/authentication/api_keys.ts

```diff
@@ -63,3 +104,27 @@ export async function createAPIKey({
 
   return key;
 }
+
+export async function invalidateAPIKey({
+  body,
+  loggers,
+  callAsCurrentUser,
+  isSecurityFeatureDisabled,
+}: InvalidateAPIKeyOptions): Promise<InvalidateAPIKeyResult | null> {
+  const logger = loggers.get('api-keys');
+
+  if (isSecurityFeatureDisabled()) {
+    return null;
+  }
+
+  logger.debug('Trying to invalidate an API key');
+
+  // User needs `manage_api_key` privilege to use this API
+  const result = (await callAsCurrentUser('shield.invalidateAPIKey', {
```

Comment:
__suggestion:__ it would be great if we could log all errors right here (maybe just `debug` + `error` only for unexpected ones if this method throws) so that we can easily filters logs using `api-keys` tag/context. Otherwise we'll have to rely on all consumers to properly log failures or fall back to Elasticsearch logs and try to match timestamps to reconstruct the full error flow during troubleshooting.